CMakeLists.txt

```diff
@@ -247,6 +247,11 @@ add_library(_lightgbm SHARED ${SOURCES})
 
 if(MSVC)
     set_target_properties(_lightgbm PROPERTIES OUTPUT_NAME "lib_lightgbm")
+    if(USE_DEBUG)
+      set_target_properties(_lightgbm PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreadedDebugDLL")
+    else()
+      set_target_properties(_lightgbm PROPERTIES MSVC_RUNTIME_LIBRARY "MultiThreadedDLL")
```

Comment:
Ooops, sorry! I'm reverting my previous approval. `MSVC_RUNTIME_LIBRARY` property is available only in >=`3.15` CMake. We need a more portable solution.
https://stackoverflow.com/a/56490614

Reply:
I remember some years ago I did the similar things in RGF.
Maybe this can help:
https://github.com/RGF-team/rgf/blob/ed59b999493df06672e82b3569d421519f65b313/RGF/CMakeLists.txt#L11-L21

Reply:
> years ago

😂 

thanks! you know what, I honestly think we might be able to drop this completely and take whatever the default is. We're already building the Python package successfully with MSVC without setting this property, right?

Reply:
> We're already building the Python package successfully with MSVC without setting this property, right?

That's true.

Reply:
Going to try this in the next day or two